contracts/strategies/morpho/MetaMorphoStrategy.sol

```diff
@@ -149,11 +187,104 @@ contract MetaMorphoStrategy is ERC4626Strategy {
-    apr = ratePerSecond * 365 days * 100;
+    // (eg 32943060 -> 32943060 * 24 * 3600 * 365 * 100 / 1e18 = 0.103% apr)
+    apr = ratePerSecond * 365 days * 100 + getRewardsApr(add, sub);
+  }
+
+  /// @notice return the additional rewards apr
+  /// @param add liquidity to add
+  /// @param sub liquidity to remove
+  /// @dev return a value multiplied by 1e18 eg for 2% apr -> 2*1e18. Add and sub params
+  /// are used off-chain to calculate the variation of the apr based on the liquidity added/removed
+  function getRewardsApr(uint256 add, uint256 sub) public view returns (uint256 apr) {
+    IMMVault _mmVault = IMMVault(address(strategyToken));
+    AprData memory _aprData = AprData(add, sub, _mmVault.withdrawQueueLength(), _mmVault.totalAssets());
+    address[] memory _rewardTokens = rewardTokens;
+    RewardData memory _rewardData;
+
+    for (uint256 i = 0; i < _rewardTokens.length; i++) {
+      _rewardData = rewardsData[_rewardTokens[i]];
+      if (_rewardData.sender == address(0) || (_rewardData.rewardToken == MORPHO && !morphoTransferable)) {
+        continue;
+      }
+      // for each market find the correct market associated the current rewardToken and calculate the apr 
+      apr += _getRewardApr(_mmVault, _aprData, _rewardData.marketId, _quoteRewards(_rewardData));
+    }
+  }
+
+  /// @notice return the additional rewards apr for a specific market
+  /// @param _mmVault metamorpho vault
+  /// @param _aprData apr data
+  /// @param _targetMarketId target market id
+  /// @param _rewardsInUnderlyings amount of rewards for the market, in underlyings
+  /// @dev return a value multiplied by 1e18 eg for 2% apr -> 2*1e18
+  /// @return apr additional rewards apr
+  function _getRewardApr(
+    IMMVault _mmVault, 
+    AprData memory _aprData,
+    bytes32 _targetMarketId,
+    uint256 _rewardsInUnderlyings
+  ) internal view returns (uint256 apr) {
+    IMorpho.Market memory _market;
+    IMorpho.Position memory _pos;
+    bytes32 _marketId;
+    uint256 _vaultShare;
+    uint256 _assetsSuppliedByVault;
+    uint256 _currPercOfAssetsForMarket;
+    uint256 _newTotalSupplyAssets;
+
+    // find the correct market associated with the current rewardToken and calculate the apr 
+    for (uint256 m = 0; m < _aprData.marketsLen; m++) {
+      _marketId = _mmVault.withdrawQueue(m);
+      if (_marketId != _targetMarketId) {
+        continue;
+      }
+      // get Morpho Blue market data
+      _market = MORPHO_BLUE.market(_marketId);
+      // get vault position data in the target market
+      _pos = MORPHO_BLUE.position(_marketId, address(_mmVault));
+      // get underlyings supplied by the vault in the target market
+      // totalSupplyShares : totalSupplyAssets = supplyShares : assetsSuppliedByVault
+      // => assetsSuppliedByVault = supplyShares * totalSupplyAssets / totalSupplyShares
+      _assetsSuppliedByVault = _pos.supplyShares * _market.totalSupplyAssets / _market.totalSupplyShares;
```

Comment:
note that `totalSupplyAssets` does not take into account the accrued interest

Reply:
Ok noted. I'm assuming that the interest will be added on new deposits / redeems or in general relatively frequently, for this reason I don't expect it to have a big impact in the overall calculations. I think it's still safe to use `totalSupplyAssets` and avoid some complexities

Reply:
Indeed i don't think it's a big deal